arangod/Replication/InitialSyncer.cpp

```diff
@@ -54,13 +54,16 @@ void InitialSyncer::startRecurringBatchExtension() {
   if (secs < 30) {
     secs = 30;
   }
+        
+  std::weak_ptr<Syncer> self(shared_from_this());
   _batchPingTimer = SchedulerFeature::SCHEDULER->queueDelay(
-      RequestLane::SERVER_REPLICATION, std::chrono::seconds(secs), [this](bool cancelled) {
-        if (!cancelled && _batch.id != 0 && !isAborted()) {
-          _batch.extend(_state.connection, _progress);
-          startRecurringBatchExtension();
-        } else {
-          _batchPingTimer.reset();
+      RequestLane::SERVER_REPLICATION, std::chrono::seconds(secs), [this, self](bool cancelled) {
+        if (!cancelled) {
+          auto syncer = self.lock();
+          if (syncer && _batch.id != 0 && !isAborted()) {
```

Comment:
I would not capture `this` here because it may no longer be valid at the time this lambda is called. We may only access members of `this` if `self.lock` returned a valid pointer. The `if` ensures that, but IMHO it does not communicate this requirement clearly enough. I would prefer to do all access via the `syncer` variable, or at least add a comment explaining the requirement.